src/main/java/com/nike/cerberus/endpoints/secret/WriteSecureData.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2017 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.endpoints.secret;
+
+import com.google.common.collect.Sets;
+import com.nike.cerberus.domain.VaultStyleErrorResponse;
+import com.nike.cerberus.service.PermissionsService;
+import com.nike.cerberus.service.SafeDepositBoxService;
+import com.nike.cerberus.service.SecureDataService;
+import com.nike.riposte.server.http.RequestInfo;
+import com.nike.riposte.server.http.ResponseInfo;
+import com.nike.riposte.util.AsyncNettyHelper;
+import com.nike.riposte.util.Matcher;
+import com.nike.riposte.util.MultiMatcher;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.SecurityContext;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+public class WriteSecureData extends SecureDataEndpointV1<Object, Object> {
```

Comment:
I might be missing it but are we checking permissions with this action?  Also, we should have some logging with principal name and ip and path